src/ucp/core/ucp_worker.c

```diff
@@ -1128,22 +1119,23 @@ ucp_worker_fill_uct_features(ucp_context_h context, uct_md_attr_t *md_attr,
         iface_params->features |= UCT_IFACE_FEATURE_AMO64;
     }
 
-    if ((context->config.ext.rndv_mode == UCP_RNDV_MODE_AUTO) ||
-        (context->config.ext.rndv_mode == UCP_RNDV_MODE_GET_ZCOPY) ||
-        (context->config.ext.rndv_mode == UCP_RNDV_MODE_GET_PIPELINE) ||
-        (context->config.ext.rndv_mode == UCP_RNDV_MODE_RKEY_PTR) ||
-        (context->config.ext.rndv_mode == UCP_RNDV_MODE_PUT_PIPELINE && has_gpu_memory)) {
+    if (context->num_mem_type_detect_mds > 0) {
+        iface_params->features |= UCT_IFACE_FEATURE_GET | UCT_IFACE_FEATURE_PUT;
+    }
+
+    if (context->config.ext.rndv_mode == UCP_RNDV_MODE_AUTO ||
+        context->config.ext.rndv_mode == UCP_RNDV_MODE_GET_ZCOPY ||
+        context->config.ext.rndv_mode == UCP_RNDV_MODE_GET_PIPELINE ||
```

Comment:
we don't need to check PIPELINE; pipeline will be used only if have GPU memory, and that is already covered by line 1123. Same for line 1134